src/ts/commands.ts

```diff
@@ -355,23 +355,46 @@ const maybeApplySuggestions = (
   }
 
   const tr = state.tr;
-  suggestionsToApply.forEach(({ from, to, text }) => {
+  suggestionsToApply.forEach(({ match, text }) => {
     if (!text) {
       return;
     }
 
-    const mappedFrom = tr.mapping.map(from);
-    const mappedTo = tr.mapping.map(to);
-    const replacementFrags = getPatchesFromReplacementText(
-      tr,
-      mappedFrom,
-      mappedTo,
-      text
-    );
+    // Ensure the replacement always begins at the first point that the old and new suggestions match.
+    const strToReplace = match.ranges.map(range => tr.doc.textBetween(range.from, range.to)).join('')
+    const indexOfFirstMatchingChar = getFirstMatchingChar(strToReplace, text);
+    if (indexOfFirstMatchingChar > 0) {
+      tr.deleteRange(match.from, match.from + indexOfFirstMatchingChar);
+    }
 
-    replacementFrags.forEach(frag =>
-      applyPatchToTransaction(tr, state.schema, frag)
-    );
+    let textCursor = 0;
+
+    // Apply the suggestion to the matched range.
+    //
+    // If the match is split into multiple ranges, attempts to preserve a reasonable split between
+    // ranges by allocating the suggestion the same number of characters as the original range. Extra
+    // characters are append.
+    //
+    // For example, given:
+    //   - the match 'ex-a-mple', where dashes denote splits in the match
+    //   - the suggestion 'ample'
+    // We get the result 'am-p-le'.
```

Comment:
Bit confused by this, should the result by a-mple ?

Reply:
It's perhaps an unfortunate example: we keep the splits where they are, and change the characters. Not clever, does the job: in this case,

```
ex-a-mple
123456789
  | |
  ^ ^

becomes

am-p-le
1234567
  | |
  ^ ^
```

This could perhaps be improved by doing something more clever: like looking at the diff to see where the strings first begin to match, and counting from there.

Reply:
Ah I see - the existing solution seems like a decent enough solution to me.